ydb/core/kqp/provider/yql_kikimr_type_ann.cpp

```diff
@@ -816,6 +816,65 @@ class TKiSinkTypeAnnotationTransformer : public TKiSinkVisitorTransformer
         return TStatus::Ok;
     }
 
+Ydb::Table::VectorIndexSettings SerializeVectorIndexSettingsToProto(const TCoNameValueTupleList& indexSettings) {
+    auto parseDistance = [] (std::string_view distance) -> auto {
+        if (distance == "cosine")
+            return Ydb::Table::VectorIndexSettings::DISTANCE_COSINE;
+        else if (distance == "manhattan")
+            return Ydb::Table::VectorIndexSettings::DISTANCE_MANHATTAN;
+        else if (distance == "euclidean")
+            return Ydb::Table::VectorIndexSettings::DISTANCE_EUCLIDEAN;
+        else 
+            YQL_ENSURE(false, "Wrong index setting distance: " << distance);
+    };
```

Comment:
Can you deduplicate this functions with almost same functions in `ydb/core/kqp/provider/yql_kikimr_exec.cpp`?

Reply:
Done